modules/extensions-mule-support/src/test/java/org/mule/runtime/module/extension/mule/internal/operation/MuleOperationProcessingStrategyTestCase.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.mule.internal.operation;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.junit.Assert.assertThat;
+import static org.mule.test.allure.AllureConstants.ReuseFeature.REUSE;
+import static org.mule.test.allure.AllureConstants.ReuseFeature.ReuseStory.OPERATIONS;
+
+import io.qameta.allure.Description;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+import org.junit.Test;
+import org.mule.extension.mule.testing.processing.strategies.test.api.ExecutionThreadTracker;
+import org.mule.functional.junit4.MuleArtifactFunctionalTestCase;
+
+import javax.inject.Inject;
+
+@Feature(REUSE)
+@Story(OPERATIONS)
+public class MuleOperationProcessingStrategyTestCase extends MuleArtifactFunctionalTestCase {
+
+  @Inject
+  private ExecutionThreadTracker executionThreadTracker;
+
+  @Override
+  protected String getConfigFile() {
+    return "mule-operation-processing-strategy-test-config.xml";
+  }
+
+  @Test
+  // .+====================================+......................................
+  // .|..Composed.Operation................|......................................
+  // .|....+=============================+.|.......+============================+.
+  // .|....|.Operation...................|.|.......|.Operation..................|.
+  // .|....|..-.ExecutionType=BLOCKING...|.|.......|..-.ExecutionType=CPU_LITE..|.
+  // .|....|..-.CompletionCallback=NO....|.|.====>.|..-.CompletionCallback=NO...|.
+  // .|....|..-.ExecutionPhase=PhaseA....|.|.......|..-.ExecutionPhase=PhaseB...|.
+  // .|....|..-.CompletionPhase=PhaseC...|.|.......|..-.CompletionPhase=PhaseD..|.
+  // .|....+=============================+.|.......+============================+.
+  // .+====================================+......................................
+  //
+  // Assertion: PhaseA == PhaseB
+  //
+  @Description("Given a composed operation with only a BLOCKING child operation, and a CPU_LITE operation without callback, " +
+      "when they are executed in that order, the BLOCKING child execution thread is the same as the CPU_LITE")
+  public void blockingOperationInsideBodyDoesNotJumpThreadAfterExecute() throws Exception {
+    flowRunner("blockingFlow").run();
+
+    Integer blockingOpExecutionPhase = getExecutionThreadPhase("Blocking child");
+    Integer nextOpExecutionPhase = getExecutionThreadPhase("After operation with one blocking child");
+    assertThat(blockingOpExecutionPhase, is(nextOpExecutionPhase));
```

Comment:
i think that this will be more readable by inlining the Integer vars and the other changes at the bottom of this class.